compiler warning in frollR.c

Installed dev data.table version now (22/12/2018 20:45:00 GMT) and got this warning:

```
/usr/local/opt/llvm/bin/clang -fopenmp -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG   -I/usr/local/opt/gettext/include -I/usr/local/opt/llvm/include  -fopenmp -fPIC  -g -O3 -Wall -pedantic -std=gnu99 -mtune=native -pipe -c frollR.c -o frollR.o
frollR.c:62:14: warning: variable 'nk' is used uninitialized whenever 'if' condition is false [-Wsometimes-uninitialized]
  } else if (badaptive) {                                       // validating n input for adaptive=TRUE
             ^~~~~~~~~
frollR.c:87:12: note: uninitialized use occurs here
  int* ikl[nk];                                                 // pointers to adaptive window width
           ^~
frollR.c:62:10: note: remove the 'if' if its condition is always true
  } else if (badaptive) {                                       // validating n input for adaptive=TRUE
         ^~~~~~~~~~~~~~~
frollR.c:42:13: note: initialize the variable 'nk' to silence this warning
  R_len_t nk;                                                   // number of rolling windows, for adaptive might be atomic to be wrapped into list
            ^
             = 0
1 warning generated.
```

--- a/src/frollR.c
+++ b/src/frollR.c
@@ -39,7 +39,7 @@ SEXP frollfunR(SEXP fun, SEXP obj, SEXP k, SEXP fill, SEXP algo, SEXP align, SEX
     error("adaptive must be TRUE or FALSE");
   bool badaptive = LOGICAL(adaptive)[0];
 
-  R_len_t nk;                                                   // number of rolling windows, for adaptive might be atomic to be wrapped into list
+  R_len_t nk = 0;                                               // number of rolling windows, for adaptive might be atomic to be wrapped into list, 0 for clang -Wall
   SEXP ik = R_NilValue;                                         // holds integer window width, if doing non-adaptive roll fun
   SEXP kl = R_NilValue;                                         // holds adaptive window width, if doing adaptive roll fun
   if (!badaptive) {                                             // validating n input for adaptive=FALSE
@@ -59,7 +59,7 @@ SEXP frollfunR(SEXP fun, SEXP obj, SEXP k, SEXP fill, SEXP algo, SEXP align, SEX
     while (i < nk && INTEGER(ik)[i] > 0) i++;
     if (i != nk)
       error("n must be positive integer values (> 0)");
-  } else if (badaptive) {                                       // validating n input for adaptive=TRUE
+  } else {                                                      // validating n input for adaptive=TRUE
     if (isVectorAtomic(k)) {                                    // if not-list then wrap into list
       kl = PROTECT(allocVector(VECSXP, 1)); protecti++;
       if (isInteger(k)) {                                       // check that k is integer vector

